### DISCUSSION TITLE
Internal Server Error,

### DISCUSSION THREAD

**Discussion by iSparkySystems** [amidaware/tacticalrmm#953] [Q&A]
Nginx first did not start up - fixed

2022/01/25 15:59:05 [emerg] 1018#1018: open() "/rmm/api/tacticalrmm/tacticalrmm/private/log/access.log" failed (2: No such file or directory)

was missing, now after install - internal server error - the same happens on Debian 10/11 and the latest Ubuntu Server.

Not to sure what to look for here? Latest install script - two seperate domains same result

**Comment by wh1te909** [comment]
please describe your environment in detail, like are you running this in your homelab or in the cloud (which vps provider?) 
please do a new install on a fresh system and use following command `bash -x install.sh 2>&1 | tee install.log`
join our discord and DM me the `install.log` thanks

**Reply by iSparkySystems** [reply]
This is homelab based Vm on Proxmox like so many others, first thought firewall filtering seems to be not the case - Right now on second domain first cert. First domain 5 certs done - paused until can surpass the let's encrypt rate limit on first domain. Verified installation steps with youtube - pretty straight forward and neat - really well done to all concerned with the installation script. Last 2 installs - 2FA has not occurred during install - certificate however has. Will need to mod that installation script to use existing certificates, as rate limit are problematic. Thank you so much for taking the time to reply. Will need time to familiarise the installation script and what to import export to make the cert work from the current vm. Will revert

**Reply by dinger1986** [reply]
Are you wiping the machine and starting the install on a fresh machine every time?

**Reply by iSparkySystems** [reply]
That is correct - currently patiently waitin on windows updates to complete before starting that install again. Almost with you - will post feedback as soon as I can. Almost with you - unfortunately no eta on that windows updates. Brilliant of you to follow up again.

**Reply by dinger1986** [reply]
windows?

**Reply by iSparkySystems** [reply]
The PC that SSH From oi

**Reply by dinger1986** [reply]
Cool

**Reply by iSparkySystems** [reply]
Tried reinstalling same result as before - cannot do anymore certs out of tries on letsencrypt for the moment. had to add tactical to the sudoers file manually - lots of step by step backups to get this far - not sure what the fix is for the moment - have not gone through the log as yet. will do shortly

**Reply by dinger1986** [reply]
@iSparkySystems I have deleted that as all your passwords are on show. I suggest you delete the file as well. Please DM it to myself or @wh1te909 on Discord as that is more secure. You need to add tactical user to sudoers as per the docs when you install, this is a manual step.

**Reply by iSparkySystems** [reply]
usermod -a -G sudo tactical the documentation followed precisely - passwords in the log are test only really do not care much who sees it server are on its own vlan with obscure ips port closed anyway.

**Reply by dinger1986** [reply]
this all sounds very similar to a chap who recently had issues on discord and it turned out to be an issue with his datastore on proxmox

**Reply by iSparkySystems** [reply]
That would not explain why mesh.domain are working the main webby not quite - looking for the answer on that part with you. We are not gonna do finger pointing here - let's find the answer go from there. after mesh setup the script exits, the why part I do not know as yet. Log to follow on discord then we take it from there

**Reply by dinger1986** [reply]
mesh doesnt rely on python, I think you are getting python errors. not finger pointing on my part, was an observation. which version of ubuntu did you try it on?

**Reply by iSparkySystems** [reply]
Description:    Ubuntu 20.04.3 LTS

**Reply by dinger1986** [reply]
weird, ok sorry I have no idea why its not working, can you try it on a different system?

**Reply by iSparkySystems** [reply]
it seems pip errors - from what little I have played around this far - Pip manually installed after going through the installation script
quasar failing at the moment - not quite sure on the fix tbh

**Reply by iSparkySystems** [reply]
already tried it on debian 10 and 11 latest version as per their website - the problem are rate limits per lets encrypt

**Reply by dinger1986** [reply]
on your proxmox host, I meant on another host

**Reply by iSparkySystems** [reply]
Thanx all sorted on the same host - python was a version lower or whatever reason - removed mongo - removed postgresql - reran install script - after doing all the manual pip installs - and fixing the path for the qr code generation. This far all the webbies appear to be working ok. Will deploy agents and mesh soon enough to test the rest. Thanx to everyone and there input

**Comment by iSparkySystems** [comment]
at the moment running through the installation script manually - received a reply on discord moments ago. Looking for the fix - the current host have many vms that installed manually just fine - they all shutdown to get this one item going. I want to understand the problem more than anything else and not bypass the problem by moving to a different hypevisor - there are options sure - does not explain the problem though. I want that problem nailed down - if it helps the project grow we do so...

**Reply by iSparkySystems** [reply]
Good Morning - changed the install script to ignore the certbot - re ran installation - now I have a web ui. However cannot login if I run /rmm/api/env/bin/python /rmm/api/tacticalrmm/manage.py reset_password <username> as user root or tactical I receive permission denied. Could you please tell me how to rectify?

**Reply by wh1te909** [reply]
i looked through the install log you sent, quasar build failed due to network connectivity looks like maybe you lost connection
you should not be running anything after install as root, only as the tactical user
doesn't appear python installed properly as there were some pip errors when trying to generate the barcode for 2fa
can't help you any further without getting onto your machine and debugging in real time so if you're down for that let me know and we can schedule something

**Reply by wh1te909** [reply]
install script works fine on proxmox too i have several production installs running on proxmox vms
